app/src/main/java/io/oldering/tvfoot/red/app/domain/matches/InfiniteScrollEventObservable.java

```diff
@@ -54,14 +54,18 @@
           totalItemCount = linearLayoutManager.getItemCount();
           firstVisibleItem = linearLayoutManager.findFirstVisibleItemPosition();
 
+          if (totalItemCount < previousTotal) {
+            previousTotal = totalItemCount;
+          }
```

Comment:
Did not really check if this was a costly check on every scroll event. Was looking ok in the emulator.